core/scripts/chaincli/handler/keeper.go

```diff
@@ -431,23 +453,67 @@ func (k *Keeper) deployUpkeeps(ctx context.Context, registryAddr common.Address,
 		// Deploy
 		var upkeepAddr common.Address
 		var deployUpkeepTx *types.Transaction
+		var checkData []byte
+		var extraData []byte
 		var err error
-		if k.cfg.UpkeepAverageEligibilityCadence > 0 {
-			upkeepAddr, deployUpkeepTx, _, err = upkeep.DeployUpkeepPerformCounterRestrictive(k.buildTxOpts(ctx), k.client,
-				big.NewInt(k.cfg.UpkeepTestRange), big.NewInt(k.cfg.UpkeepAverageEligibilityCadence),
-			)
-		} else if k.cfg.UpkeepMercury {
+		switch k.cfg.UpkeepType {
+		case Conditional:
+			checkData = []byte(k.cfg.UpkeepCheckData)
+			extraData, err = extraDataEncoder.Encode([]interface{}{0, "0x", "0x"})
+			if err != nil {
+				log.Fatal(err)
+			}
+			if k.cfg.UpkeepAverageEligibilityCadence > 0 {
+				upkeepAddr, deployUpkeepTx, _, err = upkeep.DeployUpkeepPerformCounterRestrictive(
+					k.buildTxOpts(ctx),
+					k.client,
+					big.NewInt(k.cfg.UpkeepTestRange),
+					big.NewInt(k.cfg.UpkeepAverageEligibilityCadence),
+				)
+			} else {
+				upkeepAddr, deployUpkeepTx, _, err = upkeep_counter_wrapper.DeployUpkeepCounter(
+					k.buildTxOpts(ctx),
+					k.client,
+					big.NewInt(k.cfg.UpkeepTestRange),
+					big.NewInt(k.cfg.UpkeepInterval),
+				)
+			}
+		case Mercury:
+			checkData = []byte(k.cfg.UpkeepCheckData)
+			extraData, err = extraDataEncoder.Encode([]interface{}{0, "0x", "0x"})
+			if err != nil {
+				log.Fatal(err)
+			}
 			upkeepAddr, deployUpkeepTx, _, err = mercury_upkeep_wrapper.DeployMercuryUpkeep(
 				k.buildTxOpts(ctx),
 				k.client,
 				big.NewInt(k.cfg.UpkeepTestRange),
 				big.NewInt(k.cfg.UpkeepInterval),
 				false,
 			)
-		} else {
-			upkeepAddr, deployUpkeepTx, _, err = upkeep_counter_wrapper.DeployUpkeepCounter(k.buildTxOpts(ctx), k.client,
-				big.NewInt(k.cfg.UpkeepTestRange), big.NewInt(k.cfg.UpkeepInterval),
+		case LogTrigger:
+			upkeepAddr, deployUpkeepTx, _, err = log_upkeep_counter_wrapper.DeployLogUpkeepCounter(
+				k.buildTxOpts(ctx),
+				k.client,
+				big.NewInt(k.cfg.UpkeepTestRange),
 			)
+			logTriggerConfigType := abi.MustNewType("tuple(address contractAddress, uint8 filterSelector, bytes32 topic0, bytes32 topic1, bytes32 topic2, bytes32 topic3)")
+			logTriggerConfig, err1 := abi.Encode(map[string]interface{}{
+				"contractAddress": upkeepAddr,
+				"filterSelector":  0,                                                                    // no indexed topics filtered
+				"topic0":          "0x3d53a39550e04688065827f3bb86584cb007ab9ebca7ebd528e7301c9c31eb5d", // event sig for Trigger()
+				"topic1":          "0x",
+				"topic2":          "0x",
+				"topic3":          "0x",
+			}, logTriggerConfigType)
+			if err1 != nil {
+				log.Fatal("error here 1", err1)
+			}
+			var err2 error
+			extraData, err2 = extraDataEncoder.Encode([]interface{}{1, logTriggerConfig, "0x"})
```

Comment:
nit: `extraData, err2 := extraDataEncoder.Encode([]interface{}{1, logTriggerConfig, "0x"})` could be neater. I guess you can reuse the err variable for the above err1 and err2 as well?